scripts/check.py

```diff
@@ -94,11 +94,14 @@ def _parse_clang_version(c_compiler: str) -> Optional[Version]:
     # Ubuntu 20.04: clang version 10.0.0-4ubuntu1
     # Ubuntu 18.04: clang version 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
     # Ubuntu 16.04: clang version 3.8.0-2ubuntu4 (tags/RELEASE_380/final)
+    # Debian testing (bullseye): Debian clang version 11.0.1-2
     # Built from source: clang version 9.0.1
-    assert ver_line.startswith('clang version ')
-    ver_str = ver_line.split(' ')[2]
     # Remove distribution suffix (if any) and convert to a tuple
-    return _str_to_ver(ver_str.split('-')[0])
+    assert ver_line.startswith('clang version ') \
+        or ver_line.startswith('Debian clang version')
+    ver_match = re.search(r'version (\S+?)(-.*)', ver_line)
```

Comment:
This will not match "clang version 9.0.1" (from the "# Built from source:" comment). How about `r'version ([0-9.]+)'`?

Reply:
Yes, that looks much cleaner. I forgot about the source build.
For a quick check:
```python
In [1]: import re

In [2]: vers = [ 'clang version 10.0.0-4ubuntu1',
   ...: 'clang version 6.0.0-1ubuntu2 (tags/RELEASE_600/final)',
   ...: 'clang version 3.8.0-2ubuntu4 (tags/RELEASE_380/final)',
   ...: 'Debian clang version 11.0.1-2',
   ...: 'clang version 9.0.1' ]

In [3]: [ re.search(r'version ([0-9.]+)', v).group(1) for v in vers ]
Out[3]: ['10.0.0', '6.0.0', '3.8.0', '11.0.1', '9.0.1']
```